Assets/LeapMotion/Core/Scripts/DataStructures/Pose.cs

```diff
@@ -0,0 +1,167 @@
+﻿using UnityEngine;
+
+namespace Leap.Unity {
+
+  /// <summary>
+  /// A position and rotation.
+  /// </summary>
+  [System.Serializable]
+  public struct Pose {
+
+    public Vector3    position;
+    public Quaternion rotation;
+
+    public Pose(Vector3 position, Quaternion rotation) {
+      this.position = position;
+      this.rotation = rotation;
+    }
+
+    public static Pose identity {
+      get { return new Pose(); }
+    }
+
+    /// <summary>
+    /// Returns a delta Pose such that (other Pose).Then(delta Pose) == this Pose.
+    /// </summary>
+    public Pose From(Pose other) {
+      return new Pose(this.position - other.position,
+                      this.rotation * Quaternion.Inverse(other.rotation));
+    }
+
+    /// <summary>
+    /// Accumulates the provided Pose onto this Pose. For example:
+    /// PoseB.Then(PoseA.From(PoseB)) == PoseA.
+    /// 
+    /// Position and rotation are handled independently; Pose rotations do not affect
+    /// Pose positions.
+    /// </summary>
+    public Pose Then(Pose other) {
+      return new Pose(this.position + other.position,
+                      other.rotation * this.rotation);
+    }
+
+    public bool ApproxEquals(Pose other) {
+      return position.ApproxEquals(other.position) && rotation.ApproxEquals(other.rotation);
+    }
+
+    /// <summary>
+    /// Returns a pose interpolated (Lerp for position, Slerp for rotation)
+    /// between a and b by t from 0 to 1. This method clamps t between 0 and 1; if
+    /// extrapolation is desired, see Extrapolate.
+    /// </summary>
+    public static Pose Interpolate(Pose a, Pose b, float t) {
```

Comment:
How about Lerp and LerpUnclamped to go along with how Vector3/Quaternion/friends do?

Reply:
Because it uses Quaternion.Slerp; I'm worried that calling it Lerp will cause the assumption that it's also Quaternion.Lerp-ing

Reply:
Ah I see hmmmmmm.  I'd argue that Interpolate doesn't imply spherical interpolation either :P

Reply:
modified comment to clarify that, even though it is now called Lerp, it in fact uses Quaternion.Slerp for the rotation component.